src/engine.ml

```diff
@@ -611,15 +619,24 @@ let incoming_control_client config rng session channel now op data =
               let channel_st = Established keys in
               Ok (Some ip_config, config, { channel with channel_st }, tls_out)
           | None ->
-              (* now we send a PUSH_REQUEST\0 and see what happens *)
-              push_request tls' >>| fun (tls'', out) ->
-              let channel_st = Push_request_sent (tls'', key, tls_data) in
-              (* first send an ack for the received key data packet (this needs to be
-                 a separate packet from the PUSH_REQUEST for unknown reasons) *)
-              ( None,
-                config,
-                { channel with channel_st },
-                tls_out @ [ (`Ack, Cstruct.empty); (`Control, out) ] )))
+              let pull = Config.mem Pull config in
```

Comment:
Do we otherwise support `--tls-client` without `--pull`?

Reply:
That's the question :+1:. Not sure if it's the case even for OpenVPN.

Reply:
at least the code follows the old code -- i.e. if we do not send `IV_PROTO=4` then we send a `PUSH_REQUEST` -- and follow the old style handshake.

Reply:
I think OpenVPN supports `--tls-client` without `--pull`!

       --pull This option must be used on a client which is  connecting  to  a
              multi-client  server. It indicates to OpenVPN that it should ac‐
              cept options pushed by the server, provided they are part of the
              legal  set  of  pushable options (note that the --pull option is
              implied by --client ).

              In particular, --pull allows the server to push  routes  to  the
              client,  so  you should not use --pull or --client in situations
              where you don't trust  the  server  to  have  control  over  the
              client's routing table.

Reply:
yes, see https://github.com/robur-coop/miragevpn/pull/129#discussion_r1368424228 -- if there's interest (actual deployment, example configuration), we can have such a configuration being tested. for now, I opened #133 to keep track of this